samples/snippets/fsharp/VS_Snippets_CLR_System/system.boolean.structure/fs/format3.fs

```diff
@@ -0,0 +1,40 @@
+module format3
+
+// <Snippet5>
+open System
+open System.Globalization
+
+type BooleanFormatter(culture) =
+    interface ICustomFormatter with
+        member this.Format(_, arg, formatProvider) =
+            if arg :? bool && formatProvider = this then
+                let value = arg :?> bool
+                match culture.Name with 
+                | "en-US" -> string arg
+                | "fr-FR" when value -> "vrai"
+                | "fr-FR" -> "faux"
+                | "ru-RU" when value -> "верно"
+                | "ru-RU" -> "неверно"
+                | _ -> string arg
+            else null
```

Comment:
nitpick: what do you think about this version of the code?

```suggestion
        member this.Format(_, arg, formatProvider) =
            if formatProvider <> this then null
            else
                match arg with
                | :? bool as value -> 
                    match culture.Name with 
                    | "en-US" -> string arg
                    | "fr-FR" when value -> "vrai"
                    | "fr-FR" -> "faux"
                    | "ru-RU" when value -> "верно"
                    | "ru-RU" -> "неверно"
                    | _ -> string arg
                | _ -> null
```

In your original version you had to type-cast the arg twice, but in this one we've introduced a scope where we know `value` must be bool without duplicating work.

Separately, @dsyme should we look at incorporating fantomas into this repo for formatting the samples according to the style guide?